source/common/router/config_impl.cc

```diff
@@ -136,7 +137,12 @@ class RouteActionValidationVisitor
             envoy::type::matcher::v3::HttpRequestHeaderMatchInput::default_instance())
             ->GetDescriptor()
             ->full_name());
-    if (type_url == request_header_input_name) {
+    static std::string filter_state_input_name = TypeUtil::descriptorFullNameToTypeUrl(
+        createReflectableMessage(envoy::extensions::matching::common_inputs::network::v3::
+                                     FilterStateInput::default_instance())
+            ->GetDescriptor()
+            ->full_name());
+    if (type_url == request_header_input_name || type_url == filter_state_input_name) {
```

Comment:
Seem you want to use the filter state to match route? Is there an actually scenario for that?

Reply:
I want to use all route table features with some custom input, a sample config can be found at https://github.com/envoyproxy/envoy/issues/32807

Reply:
Can I resolve it in terms of usage?